### PR TITLE
2.0

### PR DESCRIPTION
**Pull request by steven-terrana**
This is a significant Pull Request.  The repository's file structure has changed dramatically, new libraries have been added, existing libraries have been updated, steps with lifecycle hooks can be unit tested without having to import the annotation, and we finally add a README to the repo. 

# Repository Structure

## Libraries 

Previously, all pipeline libraries lived at the root of the repository.  Over time, this cluttered the repository significantly.  On top of the clutter, the `docs` directory is not a library, but you could try to load it as one since it lived at the root of the repository as well. 

This PR places all libraries under a `libraries` directory.  I think that makes a lot of sense :). 

## Unit Tests

Previously, each libraries unit tests lived in a `unit-tests` directory within each library.  The Jenkins Templating Engine does not recognize this convention.  While JTE currently only loads immediate children as steps, there is no guarantee this will always be the case.  This setup also resulted in a complex `build.gradle` file that needed to query each library to see if a `unit-tests` directory existed. 

This PR places all unit tests under the `test` directory and within subdirectories named after the library for whom the tests are for.  For example, the `a11y` library's unit tests are under `test/a11y`.  This greatly simplified the `build.gradle` file.  Now - all test resources (the steps themselves) are under `libraries` and all of the test resources are under `test`.  I think that makes a lot of sense :). 

### Testing Lifecycle Hooks

Previously, Jenkins Spock did not implicitly load the lifecycle hook annotations.  This meant to unit test a step that leveraged lifecycle hooks, you had to import the annotation in the step.  JTE specifically makes it so that this import is not required.   

This PR adds a new `JTEPipelineSpecification` that extends the `JenkinsPipelineSpecification` to implicitly load the lifecycle hook annotations so that steps can be written idiomatically and unit tests still function. 

## Documentation

The first iteration of organizing the library documentation when using Antora was to create a module for each library.  While that is a clean way to think about it, it introduces too much overhead when adding new libraries.  With this structure, you have to add a new module directory named after the library, update the nav list in `antora.yml`, add a `pages` directory, and add a `nav.adoc` for the new library modules. 

This PR consolidates all of the libraries under the `ROOT` module. Now, to add a library, you just add `<library>.adoc` to the `docs/modules/ROOT/pages` directory and update the `ROOT` module's `nav.adoc` file.  Any images or attachments for the library can be placed in subdirectories under `docs/modules/ROOT/attachments` and `docs/modules/ROOT/images`. 

# Documentation 

* A new page has been added to the docs that outlines the process and standards for adding a new library
* The repository finally has a README.adoc 

# Libraries Updated

## Kubernetes

* updated to support Helm 3.  a new SDP Image, helm, has been added to the sdp-images repository. 
* Helm 3 removes the Tiller Server.  As such, the relevant library configurations for Tiller have been removed. 

## OWASP Dependency Checker

* Instead of a comma separated list for scan targets, an array of Ant style patterns is now accepted.  This is a more intuitive configuration, and cleanly maps directly to the CLI input flags
* Instead of a comma separated list for directories to exclude, an array of Ant style patterns is now accepted.  This is a more intuitive configuration, and cleanly maps directly to the CLI input flags
* The name of the step contributed has been changed from `static_dependency_check_analysis` to `application_dependency_scan`.  It's shorter, easier to say, and maps to how we describe that step of the trusted software supply chain when talking about SDP and DevSecOps.
* The output format options have been disabled.  We will output all possible format types.  This allows the library to archive the html report and leverage Jenkins' junit step to show the results. 
* The `cvss_threshold` option of "pass" has been removed.  If you want the scan to never fail, simply omit the configuration. 

# Libraries Added 

## PyTest

A `pytest` library was added for performing unit tests with the PyTest framework.  It generates an HTML report and contributes JUnit XML.  

## Google Lighthouse 

Finally, we can deprecate the `a11y` library now that we have an alternative for performing 508 compliance scanning.  The `google_lighthouse` library contributes a `accessibility_compliance_scan` step and  takes a `url` parameter and a set of thresholds and performs a scan, evaluating against accessibility compliance, performance, best practices, and search engine optimization. 

## Sysdig Secure

A `sysdig_secure` library has been added that leverages Sysdig's inline scanner to perform container image scanning.  A PDF report is generated.  The pass/fail status of the scan is based upon policies configured in Sysdig Secure, similarly to how SonarQube's pass/fail is based upon a Quality Gate configured in the tool. 

## Terraform 

An MVP `terraform` library has been added that performs a `terraform apply` in a configured working directory.  In the future, this library will need to be updated to support persistence of the terraform state.  This backend store should be configurable.

### CODE REVIEW

**Comment by cokieffebah** [comment]
Assign me as a reviewer, I will try to review it within the next two evenings

File: build.gradle [hunk 1]
```diff
@@ -1,9 +1,5 @@
-/*
-    org.jenkins-ci.jpi allows plugin dependency resolution
-    when specifying jenkins plugin dependencies
-*/
 plugins{
```
Comment (cokieffebah) [inline build.gradle:1]: wont let me put this in the end ..
on the last line:
    //our plugin
    jenkinsTest group: 'org.jenkins-ci.plugins', name: "templating-engine", version: "1.1.1"
is this still the right version of JTE ^^

**Comment by cokieffebah** [comment]
oops still reviewing.  just wanted to leave a comment

**Comment by cokieffebah** [comment]
@steven-terrana is there anything you want targetted in this review?

File: build.gradle (hunk 1 above)
Comment (steven-terrana) [reply build.gradle:1]: from a unit test perspective I don't think it matters, but i can update to the latest JTE version (1.6)

**Comment by steven-terrana** [comment]
hey @cokieffebah - thanks for taking the time to look at this. 

I sympathize that there's a lot in this PR. 

I would appreciate it the most if you could take a look at how the repository structure has been laid out with the new `libraries`, `test`, and documentation directories and give me a sanity check that this is a better way to organize things. 

It might be better to just look at my fork and take a look at how it's laid out: https://github.com/steven-terrana/sdp-libraries

**Comment by cokieffebah** [comment]
ok, I pulled your fork and started looking at it.

**Comment by steven-terrana** [comment]
I can probably put those in the docs directory since they're used to build the docs.

**Pull request merged**